test/functional/CacheManagement/src/tests/sharedclasses/options/TestSharedCacheJavaAPI.java

```diff
@@ -103,12 +158,39 @@ public static void main(String args[]) {
 	       	
 		    cacheList = SharedClassUtilities.getSharedCacheInfo(dir, SharedClassUtilities.NO_FLAGS, false);
 		    if (cacheList == null) {
-		    	fail("SharedClassUtilities.getSharedCacheInfo failed: no cache found");
+		    	fail("SharedClassUtilities.getSharedCacheInfo failed: no cache found with dir is " + dir);
 		    }
 		    newCacheCount = cacheList.size();
-		    if ((newCacheCount == -1) || (newCacheCount != (oldCacheCount + persistentCount + nonpersistentCount + snapshotCount))) {
-		    	fail("SharedClassUtilities.getSharedCacheInfo failed: Invalid number of cache found\t" +
-		    			"expected: " + (oldCacheCount + persistentCount + nonpersistentCount + snapshotCount) + "\tfound: " + newCacheCount);
+		    
+		    if (dirGroupAccess != null) {
+		    	int tempGroupAccessCount = 0;
+		    	cacheGroupAccessList = SharedClassUtilities.getSharedCacheInfo(dirGroupAccess, SharedClassUtilities.NO_FLAGS, false);
+		    	if (cacheGroupAccessList == null) {
+		    		fail("SharedClassUtilities.getSharedCacheInfo failed: no cache found with dir is " + dirGroupAccess);
+		    	}
+		    	tempGroupAccessCount = cacheGroupAccessList.size();
+		    	if (tempGroupAccessCount == -1) {
+		    		fail("SharedClassUtilities.getSharedCacheInfo failed: Invalid number of cache found with dir is " + dirGroupAccess);
+		    	}
+		    	cacheGroupAccessNonPersistentList = SharedClassUtilities.getSharedCacheInfo(dirRemoveJavaSharedResources, SharedClassUtilities.NO_FLAGS, false);
+		    	if (cacheGroupAccessNonPersistentList == null) {
+			    	fail("SharedClassUtilities.getSharedCacheInfo failed: no cache found with dir is " + dirRemoveJavaSharedResources);
+			    }
+		    	if (tempGroupAccessCount == -1) {
```

Comment:
Why checking tempGroupAccessCount again ? I believe it can be removed.

Reply:
correct, this should be removed too. updated.